api/v1alpha1/spacerequest_types.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SpaceRequestSpec defines the desired state of Space
+// +k8s:openapi-gen=true
+type SpaceRequestSpec struct {
+	// TierName is a required property introduced to retain the name of the tier
+	// for which this Space is provisioned
+	// If not set then the tier name will be set automatically
+	// +optional
+	TierName string `json:"tierName,omitempty"`
+
+	// TargetClusterLabels one or more labels that define a set of clusters
+	// where the Space can be provisioned.
+	// +optional
+	TargetClusterLabels map[string]string `json:"targetClusterLabels,omitempty"`
```

Comment:
I would use just an array of strings instead of map
```suggestion
	TargetClusterLabels []string `json:"targetClusterLabels,omitempty"`
```
The labels at ToolchainClusters will be defined in the format:
```
toolchain.dev.openshift.com/cluster-label/workloads: ""
```
so we can have multiple labels defined for one CR. The actual value doesn't matter - what matters is the key of the label.
I took the inspiration from how k8s defines types of nodes: master, worker, or infra. They use the format:
```
node-role.kubernetes.io/worker: ''
```

So, if we don't care about the values, then I would just accept an array of strings that define the set of label keys it should match. 

cc @alexeykazakov & @rajivnathan as they were part of the discussion in the SpaceRequest document as well.
WDYT?

Reply:
My only point in favor of the map it's that sometimes it's easier to check if it has specific string keys.

For example checking if a cluster is workload type one would be something like :

```
if _, found := m["toolchain.dev.openshift.com/cluster-label/workloads"]; found {
    fmt.Println("is workload type cluster")
}
```

but I don't have strong opinions on map vs slice.
If the standard in k8s libraries is to use slices we should go with that I agree. They may have good reasons for it.

Reply:
Yeah... I like the array option better. And if we go with this option then let's rename `TargetClusterLabels` to `TargetClusterLabelKeys` to make it crystal clear that it's about the label keys. Not the entire label with key/value.

Reply:
👍  I've changed to slice of strings and renamed it, see : [dcad81e788db5a](https://github.com/codeready-toolchain/api/pull/333/commits/dcad81e788db5a4c2a560b998f263082c577f558) 

thanks for the feedback!

Reply:
Having a map of key/value pairs now instead of an array of keys could be helpful if in the future we also want to check the values, and it's doesn't make such a big difference in the model now. But changing it later will be trickier 🤷‍♂️

Reply:
Good idea with the name `TargetClusterLabelKeys`

I see your point @xcoulon. However, since this is a "public-facing" API, I would love to have it simple - that's why I'm proposing the array. It will be other StoneSoup controllers who will create the CRs, not us. The fact that we store it as a label (key-value pair) is just our implementation detail. 
We could also name the parameter `TargetClusterTaints`, `TargetClusterRoles`, or `TargetClusterAPITypes`.

And if we would need to (for any reason) change it to a map in the future, then we can easily create a new parameter with a new name. That approach will be also easier for migration on the other controller's side